runtime/runtime/src/state_viewer/mod.rs

```diff
@@ -132,10 +132,12 @@ impl TrieViewer {
                 let code_len = get_code(state_update, account_id, Some(account.code_hash()))?
                     .map(|c| c.code.len() as u64)
                     .unwrap_or_default();
-                if account.storage_usage() > Self::CONTRACT_STATE_SIZE_LIMIT + code_len {
-                    return Err(errors::ViewStateError::AccountStateTooLarge {
-                        requested_account_id: account_id.clone(),
-                    });
+                if let Some(limit) = self.state_size_limit {
+                    if account.storage_usage() - code_len > limit {
```

Comment:
Or move it to the right
```suggestion
                    if account.storage_usage().saturating_sub(code_len) > limit {
```